<launch>
	<!-- global rosparam -->
	<param name="use_sim_time" value="true"/>

	<!-- rosbag -->
	<node pkg="rosbag" type="play" name="player" args="--clock /home/amsl/ozaki/bagfiles/stick_vlp32imu4cam/with_vicon/20201213_183636.bag"/>

	<!-- rviz -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find attitude_estimation_walls)/rviz_config/attitude_estimation_walls.rviz"/>

	<!-- velodyne -->
	<include file="$(find velodyne_pointcloud)/launch/VLP-32C_points.launch"/>

	<!-- MAIN -->
	<node pkg="attitude_estimation_walls" name="wall_normal_estimation" type="wall_normal_estimation" output="screen">
		<remap from="/cloud" to="/velodyne_points"/>
		<remap from="/quat" to="/ekf/quat_rpy"/>
		<!-- <param name="mode_open_viewer" type="bool" value="false"/> -->
		<param name="omp_num_threads" type="int" value="3"/>
		<param name="skip" type="int" value="3"/>
		<param name="search_radius_ratio" type="double" value="0.09"/>
		<param name="min_search_radius" type="double" value="0.1"/>
		<param name="th_num_neighborpoints" type="int" value="10"/>
		<param name="th_fitting_error" type="double" value="0.05"/>
	</node>
	<node pkg="attitude_estimation_walls" name="dgsphere_to_gravity" type="dgsphere_to_gravity" output="screen">
		<remap from="/cloud" to="/dgsphere/selected"/>
		<remap from="/quat" to="/ekf/quat_rpy"/>
		<param name="min_num_cluster_members" type="int" value="20"/>
	</node>
	<node pkg="attitude_estimation_walls" name="ekf_gyro_dgsphere" type="ekf_gyro_dgsphere" output="screen">
		<param name="wait_inipose" type="bool" value="false"/>
		<param name="sigma_lidar_g" type="double" value="5.0e+0"/>
	</node>

	<!-- evaluation -->
	<include file="$(find attitude_estimation_walls)/launch/comparison_with_gt_real.launch"/>
</launch>
